---
title: "ABCC_Coursework"
format: html
editor: visual
---

# Group Project ABCC

## Obtaining Raw Data

### Downloading Raw Sequencing Files

```{bash}
#make directory to downoad FASTQ files into
mkdir /scratch/grp/msc_appbio/Group8_ABCC

#downloading raw sequencing files from NCBI Expression Omnibus (accession no.GSE54825)
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR116/003/SRR1166443/SRR1166443.fastq.gz 
#repeated for each of six FASTQ files: SRR1166442, SRR1166443, SRR116644, SRR1166445, SRR1166446, SRR1166447

#unzipped all the files
gzip -d SRR1166447.fastq.gz 

#organised sequencing files based on their conditions
mkdir Glucose
mkdir Cellobiose
```

### Downloading Reference Genome

Downloaded from <https://www.yeastgenome.org/>

```{bash}
wget http://sgd-archive.yeastgenome.org/sequence/strains/BY4742/BY4742_Stanford_2014_JRIR00000000/BY4742_Stanford_2014_JRIR00000000.fsa.gz
```

### Creating Custom Reference

```{bash}
#downloaded N. crassa cdt-1 and gh1-1 gene sequences
wget "https://www.ncbi.nlm.nih.gov/gene/3879950" -O cdt1.fasta

nano BY4742_Toronto_2012.fsa
#manually added the two genes into our reference genome and renamed to custom_reference_genome
```

## Quality Control (QC)

### Assess Read Quality

```         
```

### Trim and Filter Reads

Trim based on the QC results

```{bash}
{bash}
#!/bin/bash

module load trimmomatic
module load fastqc

# Set directories
INPUT_DIR=/scratch/grp/msc_appbio/Group8_ABCC/Cellobiose           # Change to where your FASTQ f>
OUTPUT_DIR=/scratch/grp/msc_appbio/Group8_ABCC/Cellobiose/trimmed_changed_results
FASTQC_POST=/scratch/grp/msc_appbio/Group8_ABCC/Cellobiose/fastqc_post_changed

# Create necessary directories
mkdir -p "$OUTPUT_DIR"
mkdir -p "$FASTQC_POST"

echo "Starting trimming for all samlpes"

# SRR1166445 - Poor quality at ends + N content
echo "Trimming SRR1166445..."
trimmomatic SE -phred33 \
$INPUT_DIR/SRR1166445.fas tq \
$OUTPUT_DIR/
SRR1166445_trimmed.fastq \
HEADCROP:4 \
CROP:35 \
LEADING:5 \
TRAILING:5 \
SLIDINGWINDOW:4:20 \
MINLEN:25

echo "SRR1166445 complete."

# SRR1166446 - Poor quality at ends
echo "Trimming SRR1166446..."
trimmomatic SE -phred33 \
$INPUT_DIR/SRR1166446.fas tq \
$OUTPUT_DIR/SRR1166446_trimmed.fastq \
HEADCROP:4 \
CROP:35 \
LEADING:5 \
TRAILING:5 \
SLIDINGWINDOW:4:20 \
MINLEN:25

#SRR1166447 
echo "Trimming SRR1166447..."
trimmomatic SE -phred33 \
$INPUT_DIR/SRR1166447.fastq \
$OUTPUT_DIR/SRR1166447_trimmed.fastq \
HEADCROP:4 \
LEADING:3 \
TRAILING:3 \
SLIDINGWINDOW:4:20 \
MINLEN:30

echo"SRR1166447 complete"

echo "All trimming complete! Running FastQC on trimmed reads..."
fastqc $OUTPUT_DIR/*_trimmed.fastq -o $FASTQC_POST -t 3
echo "=========================================="
echo "Trimming Summary:"
echo "SRR1166445: Trimmed first 4bp, kept bp 5-35 (~31bp reads)"
echo "SRR1166446: Trimmed first 4bp, kept bp 5-35 (~31bp reads)"
echo "SRR1166447: Trimmed first 4bp, light quality filtering (~46bp reads)"
echo "=========================================="
echo "Check FastQC results in: $FASTQC_POST"
```
